## Samples

The data is from 4 people, 1 healthy person and 3 chronic fatigue syndrome (CFS) patients.
From each person, 2 samples are studied: 1) total peripheral blood cells (TA) and 2) CD4+ peripheral blood cells (CD4).
In total, there are 8 samples.

## Gene-by-Sample

The input data is 16 FASTQ files, forward and reverse reads per sample, generated by paired-end RNA sequencing.
Kallisto was used to map the reads to human transcripts, ultimately resulting in transcript-by-sample matrix.
Transcripts were merged into gene symbols, resulting in gene-by-sample matrix.
Genes whose expression values did not change across samples were removed because they do not help comparing samples.
Expression values were shifted to set the minimum to be 1 and then logged with base 2.

The final gene-by-sample matrix is `output/number_feature_sample.tsv`.

## Comparisons

Throughout this analysis, 3 comparisons were done.

1. 2 healthy samples vs 6 CFS samples
2. 4 TA samples vs 4 CD4 samples.
3. 3 CFS TA samples vs 3 CFS CD4 samples.

## Compare Genes

Gene expression values were compared with 2 functions: 1) log ratio and 2) information correlation.
`output/compare_feature/` contains 3 directories, each for a comparison.
Within a comparison directory are 2 directories, each for a comparision function.
`function_heat_map.html` is the resulting figure, which shows the top and bottom genes (genes whose expressions are most differing across the samples being compared).
Double click the file open it in a web browser.
It is an interactive plot.
For example, `output/compare_feature/cfs_ta_vs_cfs_cd4/get_log_ratio/function_heat_map.html` shows that SELL is the most increased gene with log ratio of 2.27, and FABP5P7 is the most decreased gene with log ratio of -2.41.

Each `function_heat_map.html` is accompanied with `statistic.tsv`, which shows the scores for the rest of the genes.

## Gene Set Enrichment Analysis (GSEA)

Gene-by-sample matrix was converted into gene-set-by-sample matrix using GSEA and the MSigDB gene sets, covering over 20,000 pathways, ontologies, and more.

The gene-set enrichment scores were compared, and the results are organized similarly as above in `output/compare_feature/`.
Because computing log ratio across the enrichment score does not make sense, comparison was done using only information correlation.
For example, `output/compare_set/cfs_ta_vs_cfs_cd4/get_ic/function_heat_map.html` shows the top and bottom gene sets from all of the gene sets.

To compare gene sets across each gene-set collection, separate figues using gene sets from a specific collection was made.
For example, `output/compare_set/cfs_ta_vs_cfs_cd4/get_ic/ipa.gmt/function_heat_map.html` shows the most differing Ingenuity pathways.

## Interpretation

### CFS

Looking at `compare_set/cfs/get_ic/c8.all.v7.4.symbols.gmt/function_heat_map.html`:
C8 collection contains gene sets derived from single cell experiments.
CFS samples have more effector memory T cells and some strange cell populations (ovarian cells and kidney cells) and less olfactory neuroepithelium T cells.

Looking at `compare_set/cfs/get_ic/gene_set.tsv/function_heat_map.html`:
CFS samples have more CFS signals.

Looking at `compare_set/cfs/get_ic/ipa.gmt/function_heat_map.html`:
CFS samples have more ELF4 activities.

![](image.png)

### CD4+ from all

Looking at `compare_feature/`:
It is strange that CD4 expression lower in the non-CD4 samples.
This is the case for both comparison 1 and 3.

Looking at `compare_set/cd4/get_ic/`:
CD4 samples have more mature T cell signals.

Looking at `compare_set/cd4/get_ic/c8.all.v7.4.symbols.gmt/function_heat_map.html`:
CD4 samples have more lung CD4 naive T cells.

Looking at `compare_set/cd4/get_ic/h.all.v7.4.symbols.gmt/function_heat_map.html`:
CD4 samples have more MYC signals.

![](image.png)

### CD4+ from CFS

Looking at `compare_set/cfs/get_ic/gene_set.tsv/function_heat_map.html`:
The smaller CD4 signature from the celiac disease paper is higher in the CD4 samples but the larger one is lower.
This could be due to the large size of the signature and the differences in the disease context.

The trends seen in the above CD4 comparison is similar in this CFS CD4 comparison.

![](image.png)

### Summary

`output/summarize/` contains the figures that summrizes these interpretations.
